### PR TITLE
JY Create Beer Carousel

### PR DESCRIPTION
**Pull request by jenbyte**

### CODE REVIEW

File: client/js/components/Availability/styles.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { StyleSheet } from 'react-native';
+import { center, colors, fonts, row } from '../../config/styles';
+
+const styles = StyleSheet.create({
+  container: {},
```
Comment (siddharthparmar7) [inline client/js/components/Availability/styles.js]: remove container.

File: client/js/components/BeerCarousel/BeerCarousel.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
+import React, { Component } from 'react';
+import { Dimensions, Image, Text, View, TouchableOpacity } from 'react-native';
+import Carousel from 'react-native-snap-carousel';
+
+import styles from './styles';
+import { hr, center, h2, h3, Subtitle1 } from '../../config/styles';
+
+// onPress={() => {
+//   this._carousel.snapToNext();
+// }}
```
Comment (siddharthparmar7) [inline client/js/components/BeerCarousel/BeerCarousel.js]: remove commented code.

File: client/js/components/BeerCarousel/BeerCarousel.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+export class BeerCarousel extends Component {
+  _renderItem({ item, index }) {
```
Comment (siddharthparmar7) [inline client/js/components/BeerCarousel/BeerCarousel.js]: if you are using _ for your methods then you should follow that convention for all your methods, throughout your project.

File: client/js/components/BeerCarousel/BeerCarousel.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      slider1ActiveSlide: SLIDER_1_FIRST_ITEM
+    };
+  }
+
+  _renderItem({ item, index }) {
```
Comment (siddharthparmar7) [inline client/js/components/BeerCarousel/BeerCarousel.js]: same as above.

File: client/js/components/BeerCarousel/BeerCarousel.js [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    return <SliderEntry data={item} even={(index + 1) % 2 === 0} />;
+  }
+
+  _renderItemWithParallax({ item, index }, parallaxProps) {
```
Comment (siddharthparmar7) [inline client/js/components/BeerCarousel/BeerCarousel.js]: same as above.

File: client/js/components/BeerCarousel/BeerCarousel.js [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+      <SliderEntry
+        data={item}
+        even={(index + 1) % 2 === 0}
+        parallax={true}
+        parallaxProps={parallaxProps}
+      />
+    );
+  }
+
+  _renderLightItem({ item, index }) {
```
Comment (siddharthparmar7) [inline client/js/components/BeerCarousel/BeerCarousel.js]: same as above

File: client/js/components/BeerCarousel/BeerCarousel.js [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+    return <SliderEntry data={item} even={false} />;
+  }
+
+  _renderDarkItem({ item, index }) {
```
Comment (siddharthparmar7) [inline client/js/components/BeerCarousel/BeerCarousel.js]: same as above.

File: client/js/components/BeerCarousel/BeerCarousel.js [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+    return <SliderEntry data={item} even={true} />;
+  }
+
+  mainExample(number, title) {
```
Comment (siddharthparmar7) [inline client/js/components/BeerCarousel/BeerCarousel.js]: looks like you copied this file from the docs and didn't remove unnecessary code. Please make the changes.

File: client/js/components/BeerCarousel/BeerCarousel.js [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+        <Text>WHATSUPPPPPP</Text>
+        {/* <Carousel
+          ref={c => {
+            this._carousel = c;
+          }}
+          data={this.state.entries}
+          renderItem={this._renderItem}
+          sliderWidth={sliderWidth}
+          itemWidth={itemWidth}
+        /> */}
```
Comment (siddharthparmar7) [inline client/js/components/BeerCarousel/BeerCarousel.js]: remove unnecessary code and comments.

File: client/js/screens/AllBeers/AllBeersContainer.js [hunk 10]
```diff
@@ -1,8 +1,10 @@
...
+import { Text } from 'react-native';
+import { Query } from 'react-apollo';
+import gql from 'graphql-tag';
 
 import AllBeers from './AllBeers';
-import styles from './styles';
+import Loader from '../../components/Loader';
 
 class AllBeersContainer extends Component {
   constructor(props) {
```
Comment (siddharthparmar7) [inline client/js/screens/AllBeers/AllBeersContainer.js]: remove the constructor if you are not creating a state.

File: client/js/screens/AllBeers/AllBeersContainer.js [hunk 11]
```diff
@@ -14,7 +16,32 @@ class AllBeersContainer extends Component {
...
+              abv
+              releaseDate
+            }
+          }
+        `}
+      >
+        {({ loading, error, data }) => {
+          if (loading) return <Loader />;
+          if (error) return <Text>{`Error! ${error.message}`}</Text>;
+          console.log('ERROR:', error);
```
Comment (siddharthparmar7) [inline client/js/screens/AllBeers/AllBeersContainer.js]: remove this console log.

File: client/js/screens/Event/EventContainer.js [hunk 12]
```diff
@@ -1,10 +1,10 @@
 import React, { Component } from 'react';
 import { View } from 'react-native';
 
-import Events from './Events';
+import Event from './Event';
 import styles from './styles';
 
-class EventsContainer extends Component {
+class EventContainer extends Component {
   constructor(props) {
```
Comment (siddharthparmar7) [inline client/js/screens/Event/EventContainer.js]: remove this constructor if you are not creating a state.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]

**Comment by jenbyte** [comment]
Updated

**Comment by jenbyte** [comment]
Start a new pull request with the noted changes

**Pull request closed without merging**